pkg/snapshot/filesystem_lvp.go

```diff
@@ -0,0 +1,318 @@
+package snapshot
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/image"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	kotsadmtypes "github.com/replicatedhq/kots/pkg/kotsadm/types"
+	types "github.com/replicatedhq/kots/pkg/snapshot/types"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	corev1 "k8s.io/api/core/v1"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	FileSystemLVPConfigMapName = "kotsadm-fs-lvp"
+)
+
+func DeployFileSystemLvp(ctx context.Context, clientset kubernetes.Interface, deployOptions FileSystemDeployOptions, registryConfig kotsadmtypes.RegistryConfig) error {
+	veleroNamespace, err := DetectVeleroNamespace(ctx, clientset, deployOptions.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "failed to detect velero namespace")
+	}
+
+	// Ensure the config map exists for the plugin
+	if err = EnsureLocalVolumeProviderConfigMaps(deployOptions, veleroNamespace); err != nil {
+		return errors.Wrap(err, "failed to configure local volume provider plugin config map")
+	}
+
+	return nil
+}
+
+func ValidateFileSystemDeployment(ctx context.Context, clientset kubernetes.Interface, deployOptions FileSystemDeployOptions, registryConfig kotsadmtypes.RegistryConfig) (bool, bool, error) {
+	// configure fs directory/mount. This is a legacy check to see if this directory was migrated from Minio and has an intermediate directory
+	_, hasMinioConfig, writable, err := shouldResetFileSystemMinioMount(ctx, clientset, deployOptions, registryConfig)
+	if err != nil {
+		return false, false, errors.Wrap(err, "failed to check if should reset file system mount")
+	}
+	return hasMinioConfig, writable, nil
+}
+
+func GetCurrentLvpFileSystemConfig(ctx context.Context, namespace string) (*types.FileSystemConfig, error) {
+	bsl, err := FindBackupStoreLocation(ctx, namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find velero backup storage location")
+	}
+	if bsl != nil {
+		// backup storage location exists, get file system config from there
+		switch bsl.Spec.Provider {
+		case SnapshotStoreHostPathProvider:
+			fileSystemConfig := &types.FileSystemConfig{}
+			hostPath := bsl.Spec.Config["path"]
+			fileSystemConfig.HostPath = &hostPath
+			return fileSystemConfig, nil
+		case SnapshotStoreNFSProvider:
+			fileSystemConfig := &types.FileSystemConfig{
+				NFS: &types.NFSConfig{
+					Path:   bsl.Spec.Config["path"],
+					Server: bsl.Spec.Config["server"],
+				},
+			}
+			return fileSystemConfig, nil
+		}
+	}
```

Comment:
Should there be a return here if bsl is not nil but neither of the cases match?

Reply:
no, cuz we wanna try getting the config from the configmap

Reply:
hmm, maybe actually, idk. i'll think about it